### PR TITLE
[Post Ray 2.2.0 Release] Update Ray versions to Ray 2.2.0

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
Signed-off-by: Dmitri Gekhtman <dmitri.m.gekhtman@gmail.com>

## Why are these changes needed?

This PR updates Ray versions to Ray 2.2.0.
It should be merged after the Ray 2.2.0 release.

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by kevin85421 — APPROVED** [review]
The 26 files with changes look good to me, but I did not check whether all related files are updated.

**Comment by kevin85421** [comment]
> I did my best with grep. If anything is missing, we can fix it later.

SGTM

**Review by Jeffwan — APPROVED** [review]

**Comment by xwjiang2010** [comment]
Hey folks, ray 2.2.0 release is out. The PR is unblocked to be merged if it's ready.

**Comment by DmitriGekhtman** [comment]
> Hey folks, ray 2.2.0 release is out. The PR is unblocked to be merged if it's ready.

Thanks for the update!

File: .github/workflows/test-job.yaml [hunk 1]
```diff
@@ -320,6 +320,26 @@ jobs:
...
+        uses: actions/checkout@v2
+        with:
+          # When checking out the repository that
+          # triggered a workflow, this defaults to the reference or SHA for that event.
+          # Default value should work for both pull_request and merge(push) event.
+          ref: ${{github.event.pull_request.head.sha}}
+
+      - uses: ./.github/workflows/actions/compatibility
+        with:
+          ray_version: 2.2.0
```
Comment (DmitriGekhtman) [inline .github/workflows/test-job.yaml:341]: @kevin85421 @architkulkarni @sihanwang41 @wilsonwang371 @Jeffwan 
I recommend testing compatibility of the last three 2.x.x releases _and_ the last 1.x.x.
So I've added 2.2 without removing 1.13.

**Comment by DmitriGekhtman** [comment]
I've resolved conflicts. This should be fine to merge after the CI finishes.

File: .github/workflows/test-job.yaml (hunk 1 above)
Comment (DmitriGekhtman) [reply .github/workflows/test-job.yaml:341]: I recommended noting the need to update this compatibility matrix in the Anyscale-internal Ray release instructions.

Comment (architkulkarni) [reply .github/workflows/test-job.yaml:341]: Sounds good, I've added the note in the instructions and linked to this thread.

**Pull request merged**